nifi-nar-bundles/nifi-kudu-bundle/nifi-kudu-processors/src/main/java/org/apache/nifi/processors/kudu/ScanKudu.java

```diff
@@ -0,0 +1,398 @@
+        }
+
+        @Override
+        public FlowFile getFlowFile(){
+            return flowFile;
+        }
+
+        public long getRecordsCount(){
+            return rowsPulledHolder.get();
+        }
+
+    }
+
+
+    /*
+     * Initiates FF content, adds relevant attributes, and starts content with JSON array "["
+     */
+    private FlowFile initNewFlowFile(final ProcessSession session, final FlowFile origFF, final String tableName) throws IOException{
+
+        FlowFile flowFile = session.create(origFF);
+        flowFile = session.putAttribute(flowFile, KUDU_TABLE_ATTR, tableName);
+        flowFile = session.putAttribute(flowFile, CoreAttributes.MIME_TYPE.key(), "application/json");
+
+        final AtomicReference<IOException> ioe = new AtomicReference<>(null);
+        flowFile = session.write(flowFile, (out) -> {
+            try{
+                out.write("[".getBytes());
+            }catch(IOException e){
+                ioe.set(e);
+            }
+        });
+
+        if (ioe.get() != null){
+            throw ioe.get();
+        }
+
+        return flowFile;
+    }
+
+    private void finalizeFlowFile(final ProcessSession session, FlowFile flowFile, final String tableName,
+                                  Long rowsPulled, Exception e) {
+        Relationship rel = REL_SUCCESS;
+        flowFile = session.putAttribute(flowFile, KUDU_ROWS_COUNT_ATTR, rowsPulled.toString());
+
+        final AtomicReference<IOException> ioe = new AtomicReference<>(null);
+        flowFile = session.append(flowFile, (out) -> {
+            try{
+                out.write("]".getBytes());
+            } catch(IOException ei) {
+                ioe.set(ei);
+            }
+        });
+        if (e != null || ioe.get() != null) {
+            flowFile = session.putAttribute(flowFile, "scankudu.error", (e==null?e:ioe.get()).toString());
+            rel = REL_FAILURE;
+        } else {
+            session.getProvenanceReporter().receive(flowFile, tableName);
+        }
+        session.transfer(flowFile, rel);
+    }
+
+    private void addPredicate(KuduScanner.KuduScannerBuilder scannerBuilder, KuduTable kuduTable, String column, Object value, KuduPredicate.ComparisonOp comparisonOp) {
+        ColumnSchema columnSchema = kuduTable.getSchema().getColumn(column);
+        KuduPredicate predicate = KuduPredicate.newComparisonPredicate(columnSchema, comparisonOp, value);
+        scannerBuilder.addPredicate(predicate);
+    }
+
+    protected void scan(ProcessContext context, ProcessSession session, KuduTable kuduTable, String predicates, List<String> projectedColumnNames, ResultHandler handler) throws Exception {
+        KuduScanner.KuduScannerBuilder scannerBuilder = this.kuduClient.newScannerBuilder(kuduTable);
+        final String[] arrayPredicates = (predicates == null || predicates.isEmpty() ? new String[0] : predicates.split(","));
+
+        for(String column : arrayPredicates){
+            if (column.contains("=")) {
```

Comment:
The ordering of these if-else statements should be changed.
`column.contains("=")` will match for example "`column1<=10"` in which case after splitting along `=` the column name will be considered to be `"column1<"` which will lead to an exception.

This way I think it would work:
```java
            if(column.contains(">=")) {
                final String[] parts = column.split(">=");
                addPredicate(scannerBuilder, kuduTable, parts[0], parts[1], KuduPredicate.ComparisonOp.GREATER_EQUAL);
            } else if(column.contains("<=")) {
                final String[] parts = column.split("<=");
                addPredicate(scannerBuilder, kuduTable, parts[0], parts[1], KuduPredicate.ComparisonOp.LESS_EQUAL);
            } else if (column.contains("=")) {
                final String[] parts = column.split("=");
                addPredicate(scannerBuilder, kuduTable, parts[0], parts[1], KuduPredicate.ComparisonOp.EQUAL);
            } else if(column.contains(">")) {
                final String[] parts = column.split(">");
                addPredicate(scannerBuilder, kuduTable, parts[0], parts[1], KuduPredicate.ComparisonOp.GREATER);
            } else if(column.contains("<")) {
                final String[] parts = column.split("<");
                addPredicate(scannerBuilder, kuduTable, parts[0], parts[1], KuduPredicate.ComparisonOp.LESS);
            }
```